apps/docs/src/content/learn/basics/app-structure.mdx

```diff
@@ -0,0 +1,94 @@
+---
+category: Basics
+title: App Structure
+order: -900
+---
+
+Threlte is making heavy use of [Svelte's Context
+API](https://svelte.dev/tutorial/context-api) as a way to pass data through the
+component tree without having to pass props down manually at every level:
+
+```svelte title="SomeComponent.svelte"
+<script>
+  const { camera, renderer } = useThrelte()
+</script>
+```
+
+To let Threlte do its magic, we recommend to follow a simple rule to structure
+your app.
+
+## Recommended App Structure
+
+The recommended app structure is to implement the component
+[`<Canvas>`](/docs/reference/core/canvas) that provides all basic contexts and
+use a single child component (typically named "Scene.svelte" in examples) for
+your Threlte app. This way, you don't need to worry about the presence of
+contexts in your components.
+
+```svelte title="App.svelte"
+<script>
+  import { Canvas } from '@threlte/core'
+  import Scene from './Scene.svelte'
+</script>
+
+<Canvas>
+  <Scene />
+</Canvas>
+```
+
+```svelte title="Scene.svelte"
+<script>
+  import { T, useFrame } from '@threlte/core'
+  import { interactivity } from '@threlte/extras'
+  import Player from './Player.svelte'
+  import World from './World.svelte'
+
+  let rotation = 0
+
+  // useFrame is relying on a context provided
+  // by <Canvas>. Because we are definitely *inside*
+  // <Canvas>, we can safely use it.
+  useFrame((_, delta) => {
+    rotation += delta
+  })
+
+  // This file is also typically the place to
+  // inject plugins
+  interactivity()
+</script>
+
+<T.Mesh rotation.y={rotation}>
+  <T.BoxGeometry />
+  <T.MeshBasicMaterial color="red" />
+</T.Mesh>
+
+<Player />
+<World />
+```
+
+## Context Not Available
```

Comment:
If this will go live with the useTask then adjust `useFrame` use

Reply:
Oh yeah, we have to change quite some stuff. It's mostly a drop-in replacement though. Also, we were able to make `useFrame` and `useRender` still work as they always did, but with the new scheduling backend.